hip-lang-config.cmake.in

```diff
@@ -68,24 +68,38 @@ include( "${CMAKE_CURRENT_LIST_DIR}/hip-lang-targets.cmake" )
 #get_filename_component cannot resolve the symlinks if called from /opt/rocm/lib/hip
 #and do three level up again
 get_filename_component(_DIR "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
+# root directory of the hip library installation
 get_filename_component(_IMPORT_PREFIX "${_DIR}/../../../" REALPATH)
 
+# extract HIP_CLANG_ROOT from CMAKE_HIP_COMPILER
+find_program(CMAKE_HIP_COMPILER_WITH_PATH ${CMAKE_HIP_COMPILER})
+cmake_path(GET CMAKE_HIP_COMPILER_WITH_PATH PARENT_PATH HIP_CLANG_BIN)
+cmake_path(GET HIP_CLANG_BIN PARENT_PATH HIP_CLANG_ROOT)
```

Comment:
This should follow the same logic in hip-config.cmake for HIP_CLANG_ROOT, except it should use `CMAKE_HIP_COMPILER` instead of `CMAKE_CXX_COMPILER`. The logic there works for both clang and hipcc. I think this will only work when setting `CMAKE_HIP_COMPILER` to clang.

Reply:
CMAKE_HIP_COMPILER never accept hipcc. Clang is required. If you pass hipcc to CMAKE_HIP_COMPILER, CMake stops you.